src/js/utils/TaskUtil.js

```diff
@@ -2,42 +2,53 @@
 import React from 'react';
 /* eslint-enable no-unused-vars */
 
+import Node from '../structs/Node';
 import Util from './Util';
-import Service from '../structs/Service';
 
 const TaskUtil = {
   /**
    * Returns a list of ips or hosts from task
    * @param  {object} task to return ip or host list from
+   * @param  {object} node to get host name from, if necessary
    * @return {Array.<string>} an array of ip addresses or hosts
    */
-  getHostList(task = {}) {
-    let {ipAddresses} = task;
-    if (ipAddresses && ipAddresses.length) {
-      return ipAddresses.map(function (address) {
-        return address.ipAddress;
-      });
+  getHostAndPortList(task = {}, node = new Node()) {
+    let networkType = this.getNetwork(task);
+    let ports = this.getPorts(task);
+    let hostName = node.getHostName();
+
+    // networkType is 'HOST', but no host name
+    if (networkType !== 'BRIDGE' && networkType !== 'USER' && !hostName) {
+      return {ports: [], hosts: []};
+    }
+
+    // networkType is 'HOST'
+    if (networkType !== 'BRIDGE' && networkType !== 'USER') {
```

Comment:
This is future proofing, and defaulting it to handle it as 'HOST' network, so if they ever add another network type, we will handle it as 'HOST' by default